src/request.js

```diff
@@ -200,6 +200,13 @@ export const getNodeRequestOptions = request => {
 	if (!headers.has('Connection') && !agent) {
 		headers.set('Connection', 'close');
 	}
+	
+	// Because of the differences in paring between Node.js's standard library and WHATWG's spec, IPv6 literal hostnames ([::1]) are treated as domain names by Node.js's standard library.
+	// Strip off the square brackets in WHATWG's URL.hostname before handing it to Node.js.
+	let hostname = parsedURL.hostname;
+	if (hostname.startsWith('[') && hostname.endsWith(']')) {
+		hostname = hostname.substring(1, hostname.length - 1)
+	}
```

Comment:
Maybe add a reference to this pull request in the comment
```suggestion
	// Because of the differences in paring between Node.js's standard library and WHATWG's spec, IPv6 literal hostnames ([::1]) are treated as domain names by Node.js's standard library.
	// Strip off the square brackets in WHATWG's URL.hostname before handing it to Node.js.
	// See https://github.com/node-fetch/node-fetch/pull/1268
	let hostname = parsedURL.hostname;
	if (hostname.startsWith('[') && hostname.endsWith(']')) {
		hostname = hostname.substring(1, hostname.length - 1)
	}
```

Reply:
@gr2m what do you think about using `https.request(url[, options][, callback])` instead of `https.request(options[, callback])` as discussed earlier https://github.com/node-fetch/node-fetch/pull/1268#issuecomment-911880484?

if we do it the other way then this part of the code isn't needed at all and should be removed (replaced with https://github.com/node-fetch/node-fetch/compare/main...serverwentdown:fix-ipv6-literal-parsing-2)
Instead the test-case could references the PR

Reply:
Oh nice I like that much better!

Reply:
I'm also in fav. of `https.request(url[, options][, callback])` then we simply just let NodeJS core handle it internally